### PR TITLE
Flatten alembic migrations

### PR DESCRIPTION
**Pull request by conbrad**
Flattens 125 Alembic migrations into 2 migrations to improve maintainability and deployment speed. Archived migrations are preserved in `alembic/versions/archive/` for historical reference. Note: production migration chain is:  `d276ba9eed1f (last pre-compression) → stamp to → 6157a8d08f28 → run migration → cf8397b26783`, the initial schema `9bb0dc8ed7fb`) and seed data (`6157a8d08f28`) migrations only apply to databases that bootstrap from scratch (e.g. dev or local). New migrations that run on production will run the same way.
  The dev deployment runs in ~3 mins now, e.g. https://github.com/bcgov/wps/actions/runs/20625237422/job/59234759443?pr=4984

  **New Migrations:**
  - `9bb0dc8ed7fb_initial_schema.py` - Creates complete database schema including partitioned tables
  - `6157a8d08f28_seed_application_data.py` - Loads 18 application data tables from compressed SQL dump (40 MB gzipped)

  **Auto-Stamping for Production:**
  - Modified `alembic/env.py` to automatically detect production databases at old revision `d276ba9eed1f`
  - Stamps to new head `6157a8d08f28` without running migrations when `ENVIRONMENT=production`
  - Prevents re-creating existing schema and data on production deployments

  **Metadata Registry:**
  - Defines a new application registry table (`application_data_tables`) for marking tables as application data  — data needed for applications to run at a baseline. These were dumped into a seed file with:
```
pg_dump --data-only --inserts --no-owner --no-privileges \
                                         -t advisory_fuel_types \
                                         -t advisory_hfi_classification_threshold \
                                         -t advisory_shape_fuels \
                                         -t advisory_shape_types \
                                         -t advisory_shapes \
                                         -t advisory_zone_status \
                                         -t combustible_area \
                                         -t fire_centres \
                                         -t fuel_type_raster \
                                         -t fuel_types \
                                         -t hfi_fire_centre_fire_start_range \
                                         -t hfi_fire_start_lookup \
                                         -t hfi_fire_start_range \
                                         -t planning_areas \
                                         -t planning_weather_stations \
                                         -t prediction_models \
                                         -t prescription_status \
                                         -t sfms_fuel_types \
                                         -t tpi_fuel_area \
                                         wps > /Users/cbrady/projects/wps/backend/packages/wps-api/alembic/data/application_seed_data.sql
```
  - Includes descriptions and seed order for each application table

**Testing locally**

  Prerequisites:
  - Local database wps (copy of production) at revision `d276ba9eed1f`
  - Virtual environment activated

  Test Steps:

  1. Verify starting state:
  `psql -d wps -c "SELECT version_num FROM alembic_version;"`
  `# Should show: d276ba9eed1f`

  2. Run auto-stamping:
  `source <root-path>/wps/backend/.venv/bin/activate`
  `ENVIRONMENT=production alembic upgrade head`

  3. Expected output:
  `✓ Production database stamped from d276ba9eed1f to 6157a8d08f28`
  `INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.`
  `INFO  [alembic.runtime.migration] Will assume transactional DDL.`

  4. Verify migration past stamped revision:
  `psql -d wps -c "SELECT version_num FROM alembic_version;"`
  `# Should show: cf8397b26783`

  5. Verify data preserved:
  `psql -d wps -c "SELECT COUNT(*) FROM advisory_shapes;"`
  `# Should show existing data intact (40 rows)`

  Testing Normal Migration Flow (Fresh Database)

  Test that migrations still work for new databases:

  1. Create fresh test database:
  `createdb wps_test`

  2. Run migrations without production flag:
  `# Update alembic.ini or env variable to point to wps_test`
  `alembic upgrade head`

  3. Expected behavior:
    - Runs `9bb0dc8ed7fb_initial_schema` (creates all tables)
    - Runs `6157a8d08f28_seed_application_data` (loads data)
    - Runs `cf8397b26783_create_application_data_tables_metadata` (adds application data table)
    - No auto-stamping occurs
  4. Verify:
  `psql -d wps_test -c "SELECT version_num FROM alembic_version;"`
  `# Should show: cf8397b26783`

  `psql -d wps_test -c "SELECT COUNT(*) FROM application_data_tables;"`
  `# Should show: 18`

  Testing Edge Cases

  Test already at head:
  `# Run again on already-stamped database`
  `ENVIRONMENT=production alembic upgrade head`
  `# Should output: (no stamping message, already at head)`

  Test wrong head revision (assertion):
  `# If someone adds a new migration after 6157a8d08f28`
  `# The assertion will fail with clear error message`

  Reset for retesting:
  `# Reset to old revision to test again`
  `psql -d wps -c "UPDATE alembic_version SET version_num = 'd276ba9eed1f';"`
# Test Links:
[Landing Page](https://wps-pr-4984-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-4984-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-4984-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-4984-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireCalc](https://wps-pr-4984-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireCalc bookmark](https://wps-pr-4984-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-4984-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-4984-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)
[SFMS Insights](https://wps-pr-4984-e1e498-dev.apps.silver.devops.gov.bc.ca/insights)
[Fire Watch](https://wps-pr-4984-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-watch)

### CODE REVIEW

File: backend/packages/wps-api/alembic/versions/9bb0dc8ed7fb_initial_schema.py [hunk 1]
```diff
@@ -0,0 +1,2045 @@
...
+        ) PARTITION BY RANGE (prediction_timestamp);
+    """)
+
+    op.execute("""
+        COMMENT ON TABLE weather_station_model_predictions IS
+        'The interpolated weather values for a weather station, weather date, and model run';
+    """)
+
+    # Create partitions
+    op.execute("""
```
[the remark this answers is not in the document]
Comment (conbrad) [reply backend/packages/wps-api/alembic/versions/9bb0dc8ed7fb_initial_schema.py]: I'm not sure it's strictly necessary, I'll take a look into this.

[the remark this answers is not in the document]
Comment (conbrad) [reply backend/packages/wps-api/alembic/versions/9bb0dc8ed7fb_initial_schema.py]: Yeah so we can mark it as a partitioned table but not partition it right away, I removed the partitions here: https://github.com/bcgov/wps/pull/4984/commits/e83f626730b8ba4283648000fe1b0af4f4acc8e6

**Comment by brettedw** [comment]
Getting an error following the instructions to do this locally, I think it has to do with adding the most recent migration

`psql -d wps -c "SELECT version_num FROM alembic_version;"`
 >version_num
 d276ba9eed1f
(1 row)


`ENVIRONMENT=production uv run alembic upgrade head`
>    assert head_revision == '6157a8d08f28', \
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Expected head revision 6157a8d08f28 but got cf8397b26783

**Comment by conbrad** [comment]
> > > cf8397b26783
> > 
> > 
> > You'll need to download and import a backup to get the correct migration hashes.
> 
> I have a local db at `d276ba9eed1f`. I might be mistaken but I think the issue may be coming from the alembic `env` file
> 
> ```python
> assert head_revision == '6157a8d08f28', \
>                 f"Expected head revision 6157a8d08f28 but got {head_revision}"
> ```

@brettedw I pointed it to the new migration here: https://github.com/bcgov/wps/pull/4984/commits/4165c30cd6891db2e749efa45c3721311324594d

**Review by dgboss — APPROVED** [review]
🔥

**Review by brettedw — APPROVED** [review]
Love it, thanks for doing this!
